File: C++/Maximum Subarray.cpp
//记录下当前sum，小于0，则对后面无贡献，归零
class Solution {
public:
    int maxSubArray(int A[], int n) {
        int sum = 0, m;
        if(n == 0) return 0;
        m = A[0];
        for(int i=0; i<n; i++) {
            sum = max(sum, 0);
            sum += A[i];
            m = max(sum, m);
        }
        return m;
    }
};

File: python/Maximum Subarray.py
class Solution:
    # @param A, a list of integers
    # @return an integer
    def maxSubArray(self, A):
        if A == None:
            return 0
        max2now = A[0]
        sum2now = 0
        for i in xrange(0, len(A)):
            sum2now = max(sum2now, 0)
            sum2now += A[i]
            max2now = max(max2now, sum2now)
        return max2now